/*
 * This file was automatically generated by EvoSuite
 * Fri Feb 12 03:43:47 GMT 2021
 */

package com.group9.bankofaz.service;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.group9.bankofaz.model.ExternalUser;
import com.group9.bankofaz.model.InternalUser;
import com.group9.bankofaz.model.Task;
import com.group9.bankofaz.model.Transaction;
import com.group9.bankofaz.model.Users;
import com.group9.bankofaz.service.RegularEmployeeImpl;
import java.util.List;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class RegularEmployeeImpl_ESTest extends RegularEmployeeImpl_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      ExternalUser externalUser0 = new ExternalUser();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.modifyExternalUserAcct(externalUser0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.viewExternalUserAcct("");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      Transaction transaction0 = new Transaction();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.authorizeTransaction(transaction0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      try { 
        regularEmployeeImpl0.cancelTransaction((Transaction) null);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Insufficient privileges to perform the action
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      Transaction transaction0 = new Transaction();
      try { 
        regularEmployeeImpl0.updateTransaction(transaction0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Insufficient privileges to perform the action
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.viewTransaction(0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.viewTransactions(" email : ");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.createTransaction((Transaction) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.setUser("");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      InternalUser internalUser0 = new InternalUser();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.updateInfo(internalUser0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.requestPrivileges("*L*0d*]$R8");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.completeTask((-2668));
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.updateTasks();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      Users users0 = new Users();
      // Undeclared exception!
      try { 
        regularEmployeeImpl0.updatePasswd(users0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.group9.bankofaz.service.RegularEmployeeImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      RegularEmployeeImpl regularEmployeeImpl0 = new RegularEmployeeImpl();
      List<Task> list0 = regularEmployeeImpl0.getTasks();
      assertNull(list0);
  }
}
